Controllers/BranchesController.cs

```diff
@@ -0,0 +1,215 @@
+            return View();
+        }
+
+        // POST: Branches/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create(BranchFormViewModel branchVM)
+        {
+            if (ModelState.IsValid)
+            {
+                var branch = _mapper.Map<Branch>(branchVM);
+
+                _context.Add(branch);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            //var branch = await _context.Branches.FindAsync(id);
+
+            if (branch == null)
+            {
+                return NotFound();
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            var branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+            return View(branchVM);
+        }
+
+        // POST: Branches/Edit/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, BranchFormViewModel branchVM)
+        {
+            if (id != branchVM.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    var branch = _mapper.Map<Branch>(branchVM);
+                    _context.Update(branch);
+                    await _context.SaveChangesAsync();
+                }
+                catch (DbUpdateConcurrencyException)
+                {
+                    if (!BranchExists(branchVM.Id))
+                    {
+                        return NotFound();
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Delete/5
+        public async Task<IActionResult> Delete(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
```

Comment:
This seems to be bug in DELETE action. you need to mark this Branch of indicated id as DELETET state and then SaveAsync _context. I would redirect to index after all (if deletion succedeed).